src/features/languageStatus.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import * as vscode from 'vscode';
+import TelemetryReporter from '@vscode/extension-telemetry';
+import { BaseLanguageClient, ClientCapabilities, FeatureState, StaticFeature } from 'vscode-languageclient';
+
+import { ExperimentalClientCapabilities } from './types';
+import * as lsStatus from '../status/language';
+
+export class LanguageStatusFeature implements StaticFeature {
+  private disposables: vscode.Disposable[] = [];
+
+  constructor(
+    private client: BaseLanguageClient,
+    private reporter: TelemetryReporter,
+    private outputChannel: vscode.OutputChannel,
+  ) {}
+
+  getState(): FeatureState {
+    return {
+      kind: 'static',
+    };
+  }
+
+  public fillClientCapabilities(capabilities: ClientCapabilities & ExperimentalClientCapabilities): void {
+    if (!capabilities['experimental']) {
+      capabilities['experimental'] = {};
+    }
+  }
+
+  public initialize(): void {
+    this.reporter.sendTelemetryEvent('startClient');
+    this.outputChannel.appendLine('Started client');
+
+    const initializeResult = this.client.initializeResult;
+    if (initializeResult === undefined) {
+      return;
+    }
+
+    const multiFoldersSupported = initializeResult.capabilities.workspace?.workspaceFolders?.supported;
+    this.outputChannel.appendLine(`Multi-folder support: ${multiFoldersSupported}`);
```

Comment:
```suggestion
```

This is unrelated to your PR, but since you're touching it here ... 

I'm not entirely sure this log message is still useful 🤔 It has been a long time since we introduced multi-folder support in LS and ~99% of users use the bundled LS, hence it's extremely unlikely this would be a differentiating factor when debugging a problem. We can still see this from the LSP request log anyway.